Update tispark-overview.md version matrix (#18251)
pingcap/docs: tispark-overview.md

| 2.5.x            | 5.x, 4.x               | 3.0.x, 3.1.x   | 2.12          |
| 3.0.x            | 5.x, 4.x               | 3.0.x, 3.1.x, 3.2.x|2.12|
| 3.1.x            | 6.x, 5.x, 4.x          | 3.0.x, 3.1.x, 3.2.x, 3.3.x|2.12|
| 3.2.x            | 6.x, 5.x, 4.x          | 3.0.x, 3.1.x, 3.2.x, 3.3.x|2.12|

TiSpark 2.4.4, 2.5.2, 3.0.2, 3.1.1, and 3.2.3 are the latest stable versions and are highly recommended.

> **Note:**
>

> > TiSpark does not guarantee the compatibility with tidb >= 7.x

Reviewer: ```suggestion
> TiSpark does not guarantee the compatibility with TiDB v7.0.0 and later versions.
```

| 2.5.x            | 5.x, 4.x               | 3.0.x, 3.1.x   | 2.12          |
| 3.0.x            | 5.x, 4.x               | 3.0.x, 3.1.x, 3.2.x|2.12|
| 3.1.x            | 6.x, 5.x, 4.x          | 3.0.x, 3.1.x, 3.2.x, 3.3.x|2.12|
| 3.2.x            | 6.x, 5.x, 4.x          | 3.0.x, 3.1.x, 3.2.x, 3.3.x|2.12|

> TiSpark 2.4.4, 2.5.2, 3.0.2, 3.1.1 and 3.2.3 are the latest stable versions and are highly recommended.

Reviewer: ```suggestion
TiSpark 2.4.4, 2.5.2, 3.0.2, 3.1.1， and 3.2.3 are the latest stable versions and are highly recommended.
```

TiSpark 2.4.4, 2.5.2, 3.0.2, 3.1.1, and 3.2.3 are the latest stable versions and are highly recommended.

> **Note:**
>

> > TiSpark does not guarantee the compatibility with TiDB v7.0.0 and later versions.

Reviewer 2: ```suggestion
> TiSpark does not guarantee compatibility with TiDB v7.0.0 and later versions.
```
